tests/conftest.py

```diff
@@ -55,9 +56,118 @@ def pytest_addoption(parser):
     )
 
 
+def _modify_compose_for_testing(request):
+    demo = "../docker-compose.demo.yml"
+    testing = "../docker-compose.testing.yml"
+    storage_proxy_demo = "../storage-proxy/docker-compose.storage-proxy.demo.yml"
+    storage_proxy_testing = "../storage-proxy/docker-compose.storage-proxy.testing.yml"
+
+    # Published ports in docker-compose-demo and docker compose.storage-proxy.demo
+    demo_ports = ["9000:9000", "8080:8080", "443:443", "80:80"]
+    demo_storage_proxy_ports = ["9000:9000"]
+
+    # Remove published ports
+    for ports, out, file in zip(
+        [demo_ports, demo_storage_proxy_ports],
+        [testing, storage_proxy_testing],
+        [demo, storage_proxy_demo],
+    ):
+
+        command = ["sed"]
+        for port in ports:
+            command.extend(["-e", f"/{port}/d"])
+        subprocess.run(command + ["-e", "/ports:/d", file], stdout=open(out, "w"))
```

Comment:
Q: will the `open` call have a corresponding `close` in this instance? :thinking: